compiler/next/test/resolution/testScopeResolve.cpp

```diff
@@ -456,16 +456,233 @@ static void test8() {
+                         "  x;\n"
+                         "  y;\n"
+                         "}\n"
+                         "module N {\n"
+                         "  var x;\n"
+                         "  var y;\n"
+                         "}\n";
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parse(context, path);
+  assert(vec.size() == 2);
+  const Module* m = vec[0]->toModule();
+  assert(m);
+  assert(m->numStmts() == 3);
+  const Module* n = vec[1]->toModule();
+  assert(n);
+  assert(n->numStmts() == 2);
+
+  const Variable* x = n->stmt(0)->toVariable();
+  assert(x);
+  const Variable* y = n->stmt(1)->toVariable();
+  assert(y);
+
+  const Identifier* xIdent = m->stmt(1)->toIdentifier();
+  assert(xIdent);
+  const Identifier* yIdent = m->stmt(2)->toIdentifier();
+  assert(xIdent);
+
+  const Scope* scopeForIdent = scopeForId(context, xIdent->id());
+  assert(scopeForIdent);
+
+  const auto& mx = findInnermostDecl(context, scopeForIdent, xIdent->name());
+  assert(mx.id == x->id());
+  assert(mx.found == InnermostMatch::ONE);
+
+  const auto& my = findInnermostDecl(context, scopeForIdent, yIdent->name());
+  assert(my.id == y->id());
+  assert(my.found == InnermostMatch::ONE);
+
+}
+
+// testing import of a nested module
+static void test12() {
+  printf("test12\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::build(context, "test12.chpl");
+  std::string contents = "module M {\n"
+                         "  import N.NN.NNN.x;\n"
+                         "  x;\n"
+                         "}\n"
+                         "module N {\n"
+                         "  module NN {\n"
+                         "    module NNN {\n"
+                         "      var x;\n"
+                         "    }\n"
+                         "  }\n"
+                         "}\n";
+  setFileText(context, path, contents);
 
+  const ModuleVec& vec = parse(context, path);
+  assert(vec.size() == 2);
+  const Module* m = vec[0]->toModule();
+  assert(m);
+  assert(m->numStmts() == 2);
+  const Module* n = vec[1]->toModule();
+  assert(n);
+  assert(n->numStmts() == 1);
+  const Module* nn = n->stmt(0)->toModule();
+  assert(nn);
+  const Module* nnn = nn->stmt(0)->toModule();
+  assert(nnn);
+  const Variable* x = nnn->stmt(0)->toVariable();
+  assert(x);
 
+  const Identifier* xIdent = m->stmt(1)->toIdentifier();
+  assert(xIdent);
+
+  const Scope* scopeForIdent = scopeForId(context, xIdent->id());
+  assert(scopeForIdent);
+
+  const auto& match = findInnermostDecl(context, scopeForIdent, xIdent->name());
+  assert(match.id == x->id());
+  assert(match.found == InnermostMatch::ONE);
+}
+
+// testing use of a nested module
+static void test13() {
+  printf("test13\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::build(context, "test13.chpl");
+  std::string contents = "module M {\n"
+                         "  use N.NN.NNN;\n"
+                         "  x;\n"
```

Comment:
This test should also verify that qualified access of x is possible as well, since use enables both

Reply:
Yeah, I can't test that today with what is in this PR (as with the above comment) but I'll add it as future work.